File: anavi-gardening-phat/led/c/blink.c
#include <wiringPi.h>
#include <stdio.h>
#include <signal.h>
#include <stdlib.h>

//Pin 15 on Raspberry Pi corresponds to BCM GPIO 3 and wiringPi pin 3
#define LED 3

void close()
{
	digitalWrite(LED, 0);
	exit(0);
}

int main()
{
	signal(SIGINT, close);

	if(-1 == wiringPiSetup())
	{
		printf("setup wiringPi failed!");
		return 1;
	}

	//Set GPIO pins
	pinMode(LED, OUTPUT);

	// Blink
	while(1)
	{
		digitalWrite(LED, 1);
		delay(500);
		digitalWrite(LED, 0);
		delay(500);
	}

	return 0;
}

File: anavi-gardening-phat/led/python/blink.py
# coding=utf-8
import signal
import sys
import time

import RPi.GPIO as GPIO

# Pin 15 on Raspberry Pi corresponds to GPIO 3
LED = 15


def close(signal, frame):
    GPIO.output(LED, 0)
    sys.exit(0)


signal.signal(signal.SIGINT, close)

# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)
GPIO.setwarnings(False)

# set up GPIO output channel
GPIO.setup(LED, GPIO.OUT)

# reset

try:
    while True:
        GPIO.output(LED, 1)
        time.sleep(0.5)
        GPIO.output(LED, 0)
        time.sleep(0.5)
finally:
    GPIO.cleanup()
